sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/application-offering-change-request.institutions.controller.ts

```diff
@@ -63,4 +68,92 @@ export class ApplicationOfferingChangeRequestInstitutionsController extends Base
       count: applications.count,
     };
   }
+
+  /**
+   * Gets all in progress application offering request changes.
+   * @param locationId location id.
+   * @param paginationOptions options to execute the pagination.
+   * @returns list of inprogress application offering request changes.
+   */
+  @HasLocationAccess("locationId")
+  @Get("in-progress")
+  async getInprogressApplications(
+    @Param("locationId", ParseIntPipe) locationId: number,
+    @Query() pagination: OfferingChangePaginationRequestedOptionsAPIInDTO,
+  ): Promise<
+    PaginatedResultsAPIOutDTO<InprogressApplicationOfferingChangesAPIOutDTO>
+  > {
+    const offeringChange =
+      await this.applicationOfferingChangeRequestService.getRequestedSummary(
+        locationId,
+        pagination,
+        [
+          ApplicationOfferingChangeRequestStatus.InProgressWithSABC,
+          ApplicationOfferingChangeRequestStatus.InProgressWithStudent,
+        ],
+      );
+    return {
+      results: offeringChange.results.map((eachOfferingChange) => {
+        const offering =
+          eachOfferingChange.application.currentAssessment.offering;
+        return {
+          applicationNumber: eachOfferingChange.application.applicationNumber,
+          applicationId: eachOfferingChange.application.id,
+          studyStartPeriod: offering.studyStartDate,
+          studyEndPeriod: offering.studyEndDate,
+          fullName: getUserFullName(
+            eachOfferingChange.application.student.user,
+          ),
+          status: eachOfferingChange.applicationOfferingChangeRequestStatus,
+        };
+      }),
+      count: offeringChange.count,
+    };
+  }
+
+  /**
+   * Gets all completed (Approved/ Declined) application offering request changes.
+   * @param locationId location id.
+   * @param paginationOptions options to execute the pagination.
+   * @returns list of completed application offering request changes.
+   */
+  @HasLocationAccess("locationId")
+  @Get("completed")
+  async getCompletedApplications(
+    @Param("locationId", ParseIntPipe) locationId: number,
+    @Query() pagination: OfferingChangePaginationRequestedOptionsAPIInDTO,
+  ): Promise<
+    PaginatedResultsAPIOutDTO<CompletedApplicationOfferingChangesAPIOutDTO>
+  > {
+    const offeringChange =
+      await this.applicationOfferingChangeRequestService.getRequestedSummary(
+        locationId,
+        pagination,
+        [
+          ApplicationOfferingChangeRequestStatus.Approved,
+          ApplicationOfferingChangeRequestStatus.DeclinedByStudent,
+          ApplicationOfferingChangeRequestStatus.DeclinedBySABC,
+        ],
+      );
+    return {
+      results: offeringChange.results.map((eachOfferingChange) => {
+        const offering =
```

Comment:
Just to check, line 140 - 150 is common for both controller methods. If there is some quick way to create a common method(private), we can do that.

Reply:
I don't think it needed, since DTOs are different for both controller